manager/integration/tests/test_ha.py

```diff
@@ -2142,8 +2142,96 @@ def test_reuse_failed_replica():
     11. Verify the failed replica (in step 5) will be reused.
     12. Verify the volume r/w still works fine.
     """
-    pass
+    long_wait = 60*60
+    short_wait = 3
+
+    replenish_wait_setting = \
+        client.by_id_setting(SETTING_REPLICA_REPLENISHMENT_WAIT_INTERVAL)
+    client.update(replenish_wait_setting, value=str(long_wait))
+
+    replica_node_soft_anti_affinity_setting = \
+        client.by_id_setting(SETTING_REPLICA_NODE_SOFT_ANTI_AFFINITY)
+    client.update(replica_node_soft_anti_affinity_setting, value="false")
+
+    vol = create_and_check_volume(client, volume_name)
+    host_id = get_self_host_id()
+    vol = vol.attach(hostId=host_id)
+    vol = common.wait_for_volume_healthy(client, volume_name)
+    data = {
+        'pos': 0,
+        'content': common.generate_random_data(16*Ki),
+    }
+    common.write_volume_data(vol, data)
+
+    current_host = client.by_id_node(id=host_id)
+    client.update(current_host, allowScheduling=False)
+
+    vol = client.by_id_volume(volume_name)
+    assert len(vol.replicas) == 3
+    other_replicas = []
+    for r in vol.replicas:
+        if r["hostId"] == current_host.name:
+            replica_1 = r
+        else:
+            other_replicas.append(r)
+    replica_2, replica_3 = other_replicas
+
+    for filenames in os.listdir(replica_1.dataPath):
+        if filenames.endswith(".img"):
+            with open(os.path.join(replica_1.dataPath, filenames), 'w') as f:
+                f.write("Longhorn is the best!")
+
+    crash_replica_processes(client, core_api, volume_name,
+                            replicas=[replica_1],
+                            wait_to_fail=False)
+
+    for i in range(30):
+        vol = client.by_id_volume(volume_name)
+        current_replica_names = set([r.name for r in vol.replicas])
+        assert current_replica_names == \
+               {replica_1.name, replica_2.name, replica_3.name}
+        time.sleep(2)
+
+    replenish_wait_setting = \
+        client.by_id_setting(SETTING_REPLICA_REPLENISHMENT_WAIT_INTERVAL)
+    client.update(replenish_wait_setting, value=str(short_wait))
+
+    new_replica = None
+    for i in range(60):
+        vol = client.by_id_volume(volume_name)
+        for r in vol.replicas:
+            if r.name not in {replica_1.name, replica_2.name, replica_3.name}:
+                new_replica = r
+        if new_replica is not None:
+            break
+        time.sleep(1)
+    assert new_replica is not None
+    assert new_replica.hostId == ""
 
+    replenish_wait_setting = \
+        client.by_id_setting(SETTING_REPLICA_REPLENISHMENT_WAIT_INTERVAL)
+    client.update(replenish_wait_setting, value=str(long_wait))
+
+    vol = client.by_id_volume(volume_name)
+    vol.replicaRemove(name=new_replica.name)
+    for i in range(30):
```

Comment:
add a comment that explains why there is an unavoidable wait here.

Reply:
Added. Thanks